### ISSUE TITLE
Forgetting to call `callback` in "After" hook ends the whole test suite

### ISSUE THREAD

**Issue by kevgo** [cucumber/cucumber-js#332]
__steps to reproduce__
* create an `After` hook that doesn't call the given `callback` parameter

__expected result:__ the suite hangs, and displays a descriptive error message after a timeout 

__actual result:__ the suite terminates after encountering that `After` block the first time

**Comment by jbpros** [comment]
Similar/related to #27.

Several cases:
- Not accepting a callback in the hook function should make it synchronous, Cucumber should consider the hook to be synchronous.
- If the function signature exposes a callback parameter, Cucumber should consider it async and time out if it never calls back.

**Comment by alwinmark** [comment]
same for the World object. Took me hours to debug why it won't work. Also if the callback will be called too late (I guess) leads to only the Feature Description output.
```
$ ./node_modules/cucumber/bin/cucumber.js -r ./features/step_definitions/common/steps.js -b
Feature: Proxy
    In Order to be able to connect my legacy Website to a new Service Landscape without touching old code
    As a (UI and Backend)-Developer
    I want to be able to Proxy (for specific paths) to my old website
```

I would expect an Error describing that the callback was never called.

**Comment by charlierudolph** [comment]
A solution to this particular problem of forgetting to execute the callback, could be setting a listener on process's 'beforeExit' event and then it can just continue with a message that the function failed to execute its callback.

**Comment by kevgo** [comment]
This NPM module could provide a nice and robust solution for the asynchronous use case. It wraps the method to be called back, and covers all problematic use cases:
* callback is never called  --> timeout
* callback is called too late   --> timeout, the late call is ignored
* callback is called several times -->  error on the second call
* in all other cases it behaves completely transparently

**Comment by jbpros** [comment]
Fixed in 2c5cd124724ba128d1d1c2b50ea80f75acef435c
